client/src/templates/Challenges/utils/frame.ts

```diff
@@ -264,9 +264,39 @@ const writeContentToFrame = (frameContext: Context) => {
     createHeader(frameContext.element.id) + frameContext.build,
     frameContext.document
   );
+
+  const iframe = document.getElementById(
+    frameContext.element.id
+  ) as HTMLIFrameElement;
+
+  registerScrollEventListner(iframe);
+
+  if (getPreviewScrollPosition()) {
+    restorePreviewScrollPosition(iframe, getPreviewScrollPosition());
+  }
   return frameContext;
 };
 
+let previewScrollPosition = 0;
+export const getPreviewScrollPosition = () => previewScrollPosition;
+export const setPreviewScrollPosition = (position: number) =>
+  (previewScrollPosition = position);
+
+const registerScrollEventListner = (iframe: HTMLIFrameElement) => {
+  iframe?.contentDocument?.addEventListener('scroll', event => {
+    const target = event.target as Document;
+    setPreviewScrollPosition(target.body.scrollTop);
```

Comment:
Two things here:
1) This type assertions suggests something fishy is going on - we should try not to assert this type (use a type guard, at the very least)
2) The `scroll` event should be rate limited: https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event#scroll_event_throttling

Reply:
@ShaunSHamilton,

`(1)` - I have removed the use of type guard.
`(2)` - Upon scroll, only the variable `previewScrollPosition` will get updated, not much of heavy operation. So, is it necessary to rate limit here?

Reply:
> not much of heavy operation. So, is it necessary to rate limit here?

You are quite right. There would probably be no benefit (_might even make it worse_ 😨 )